docs/src/main/asciidoc/spring-cloud-commons.adoc

```diff
@@ -1059,3 +1059,99 @@ myrandom=${cachedrandom.appname.value}
 == Configuration Properties
 
 To see the list of all Spring Cloud Commons related configuration properties please check link:appendix.html[the Appendix page].
+
+== Spring Cloud Request Hedging
+=== Introduction
+Request hedging is a mechanism to improve performance of idempotent requests with large worst-case response times. It
+does this by automatically retrying after the original request has been active for some configurable amount of time,
+usually based on a quantile of the request's typical response time. Unlike timing out the request and retrying, request
+hedging will keep the original requests and return the first request to succeed. This trades off a small percentage
+of additional network traffic to reduce the long-tail latency.
+
+=== Usage
+You can configure a `WebClient` to do request hedging by creating a `HedgingPolicy` `@Bean` and use the `@Hedged`
+qualifier. The `HedgingPolicy` allows you to specify how long to wait before executing the extra requests, a maximum
+number of extra requests to create. Most users will want the `LatencyPercentileHedgingPolicy`, which executes new
+requests when the previous ones have exceeded some percentile of the historical latency.
+
+This example shows a typical usage:
+
+====
+[source,java,indent=0]
+----
+@Configuration
+public class MyConfiguration {
+    @Bean
+    HedgingPolicy hedgingPolicy() {
+    	return new LatencyPercentileHedgingPolicy(
+            3, // Maximum extra requests to execute
+            0.95 // Percentile of latency to wait before executing an additional request
+    	);
+    }
+
+    @Hedged(hedgingPolicy = "hedgingPolicy")
+    @Bean
+    WebClient.Builder webClientBuilder() {
+        return WebClient.builder();
```

Comment:
So the `org.springframework.boot.autoconfigure.web.reactive.function.client.WebClientAutoConfiguration` creates a builder for us that respects the available WebClientCustomizer - we need to at least do the same. 

```java
    @Bean
    @Scope("prototype")
    @ConditionalOnMissingBean
    public Builder webClientBuilder(ObjectProvider<WebClientCustomizer> customizerProvider) {
        Builder builder = WebClient.builder();
        customizerProvider.orderedStream().forEach((customizer) -> {
            customizer.customize(builder);
        });
        return builder;
    }
```
OR
we could explore moving this annotation to a WebClientCustomizer?

Reply:
Hmm maybe it's easier/better to get rid of the `@Hedged` annotation and do something like this instead:
```
@Bean
public WebClientCustomizer hedgingWebClientCustomizer(final HedgingPolicy hedgingPolicy) {
    return HedgingWebClientCustomizer.of(hedgingPolicy);
}
```